packages/braid-design-system/src/lib/components/private/InlineField/StyledInput.tsx

```diff
@@ -153,13 +150,36 @@ export const StyledInput = forwardRef<
       ? { lightMode: 'neutralSoft', darkMode: 'neutral' }
       : { lightMode: 'surface' };
     const defaultBorder = checked ? 'formAccent' : 'default';
+    const indeterminateRef = useRef(isMixed);
+    const lastChecked = useRef<CheckboxChecked | null>(null);
+    indeterminateRef.current = isMixed;
```

Comment:
Confused about this line. Isn't the ref's initial value already set to `isMixed` on line 153?

Reply:
Line 153 is setting the default state for the ref, and then 155 is updating the ref on subsequent re-renders. Open to ways of refactoring this, but removing that fails tests (and the tests seem correct and valid)

Reply:
This makes perfect sense in hindsight 🤦